libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts

```diff
@@ -22,20 +24,34 @@ export class GenericAdrLicenseApi implements GenericLicenseClient<AdrDto> {
     private adrApi: AdrApi,
   ) {}
 
+  handleError(error: any): any {
+    this.logger.error('ADR license fetch failed', {
+      exception: error,
+      message: (error as Error)?.message,
+      category: LOG_CATEGORY,
+    })
+
+    throw new ApolloError(
+      'Failed to resolve request',
+      error?.message ?? error?.response?.message,
+    )
+  }
+
+  private handle4xx(error: FetchError) {
+    if (error.status === 403 || error.status === 404) {
```

Comment:
This could fail if `error` is falsy, better to type it as `error: Partial<Error>` or something like that?